### PR TITLE
CBL-3385: Allow null expression parameter for Function.count(expression)

### PR DESCRIPTION
**Pull request by velicuvlad**

### CODE REVIEW

**Comment by velicuvlad** [comment]
failing tests unrelated, there's a hope to address these in a another, already existing, PR.

File: Swift/Function.swift [hunk 1]
```diff
@@ -39,8 +39,12 @@ public final class Function {
     ///
     /// - Parameter expression: The expression.
     /// - Returns: The COUNT(expr) function.
-    public static func count(_ expression: ExpressionProtocol) -> ExpressionProtocol {
-        return QueryExpression(CBLQueryFunction.count(expression.toImpl()))
+    public static func count(_ expression: ExpressionProtocol? = nil) -> ExpressionProtocol {
+        if let expression = expression {
```
Comment (bmeike) [inline Swift/Function.swift:43]: I'm not sure what idiomatic Obj-C looks like so ignore this if it is irrelevant.
I would be inclined to have the value of the `let` be either expression.toImpl() or expression.property('*) and to have a single return statement.

**Review by bmeike — APPROVED** [review]
LGTM

File: Swift/Function.swift [hunk 2]
```diff
@@ -39,8 +39,12 @@ public final class Function {
...
+            return QueryExpression(CBLQueryFunction.count(expression.toImpl()))
+        } else {
+            return QueryExpression(CBLQueryFunction.count(Expression.property("*").toImpl()))
```
Comment (pasin) [inline Swift/Function.swift:46]: If expression is nil, nil can be passed to Objective-C count directory.

```
return QueryExpression(CBLQueryFunction.count(expression != nil ? expression.toImpl() : nil))
```

Comment (pasin) [reply Swift/Function.swift:46]: Need to update the `+ (CBLQueryExpression*) count: (CBLQueryExpression*)expression;` signature to allow null as well.

**Review by pasin — CHANGES_REQUESTED** [review]

**Review by pasin — APPROVED** [review]

**Pull request merged**